src/cart/crt_rpc.c

```diff
@@ -1357,6 +1357,27 @@ crt_rpc_inout_buff_init(struct crt_rpc_priv *rpc_priv)
 	}
 }
 
+static inline void
+crt_common_hdr_init(struct crt_rpc_priv *rpc_priv, crt_opcode_t opc)
+{
+	d_rank_t	rank;
+	uint32_t	xid;
+	int		rc;
+
+	rc = crt_group_rank(0, &rank);
+	D_ASSERT(rc == 0);
+
+	xid = atomic_fetch_add(&crt_gdata.cg_xid, 1);
+
+	rpc_priv->crp_req_hdr.cch_opc = opc;
+	rpc_priv->crp_req_hdr.cch_rank = rank;
+	rpc_priv->crp_req_hdr.cch_xid = xid;
```

Comment:
probably add ep_tag here too?

Reply:
ep_tag (of rpc destination) is already tracked as part of bigger rpc_priv structure

Reply:
Hmm, I mean on the server side can it know which tag the rpc is suppose to be sent?  which item in rpc_priv I can see that?

Reply:
struct crt_rpc_priv x; x.crp_pub.cr_ep.ep_tag

Reply:
ah, yes. hmm in crt_rpc_handler_common(), only ep_rank get updated ?           
                rc = crt_hg_unpack_body(rpc_priv, proc);
                if (rc == 0) {
                        rpc_priv->crp_input_got = 1;
                        rpc_pub->cr_ep.ep_rank = rpc_priv->crp_req_hdr.cch_rank;
                        rpc_pub->cr_ep.ep_grp = NULL;
                        /* TODO lookup by rpc_priv->crp_req_hdr.cch_grp_id */
                } else {

Reply:
good find. let me update code to pass along tag as well in the same manner